src/main/java/sevensmurfs/rehub/service/TherapyResultService.java

```diff
@@ -28,4 +32,24 @@ public void deleteAllTherapyResultsForTherapies(List<Therapy> therapies) {
 
         log.debug("Successfully deleted all therapy results.");
     }
+
+    @Transactional
+    public void writeTherapyResult(TherapyResultRequest request) {
+        // Find the therapy by ID
+        Therapy therapy = therapyRepository.findById(request.getTherapyId())
+                                           .orElseThrow(() -> new EntityNotFoundException("Therapy not found"));

```

Comment:
It is a good way to throw EntityNotFoundException, but if employee sets wrong therapy Id it is employees fault. So it should throw 400 Bad request which is handled with IllegalArgumentException in our solution. So it would be better to throw IllegalArgumentException.